composables/use-text-to-speech.ts

```diff
@@ -363,31 +377,36 @@ export function useTextToSpeech(options: TTSOptions = {}) {
       nft_class_id: nftClassId,
     })
     const activeAudio = audioBuffers.value[currentBufferIndex.value]
-    activeAudio?.pause()
-    playNextElement({ reset: true })
+    if (activeAudio) {
+      activeAudio.pause()
+      activeAudio.currentTime = 0
+    }
+
+    if (currentTTSSegmentIndex.value + 1 < ttsSegments.value.length) {
+      currentTTSSegmentIndex.value += 1
+    }
+    debouncedPlayback()
   }
 
+  const debouncedPlayback = useDebounceFn(() => {
```

Comment:
Want to keep a similar function name for semantic clarity
```suggestion
  const playCurrentElementDebounced = useDebounceFn(() => {
    playCurrentElement()
  }, 500)
```